extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java

```diff
@@ -262,6 +277,85 @@ public void build(
+
+            @Override
+            public boolean appliesTo(Kind kind) {
+                return kind == Kind.CLASS;
+            }
+
+            @Override
+            public void transform(TransformationContext transformationContext) {
+                ClassInfo clazz = transformationContext.getTarget().asClass();
+                if (clazz.classAnnotation(ResteasyDotNames.PATH) != null) {
+                    // Skip root resources - @Path is a bean defining annotation
+                    return;
+                }
+                if (BuiltinScope.isIn(clazz.classAnnotations())) {
+                    // Skip classes annotated with built-in scope
+                    return;
+                }
+                if (clazz.classAnnotation(ResteasyDotNames.PROVIDER) != null) {
+                    if (clazz.annotations().containsKey(DotNames.INJECT)) {
+                        // A provider with an injection point but no built-in scope is @Singleton
+                        transformationContext.transform().add(BuiltinScope.SINGLETON.getName()).done();
+                    }
+                } else if (subresources.contains(clazz.name())) {
+                    // Transform a class annotated with a request method designator
+                    transformationContext.transform()
+                            .add(resteasyConfig.singletonResources ? BuiltinScope.SINGLETON.getName()
+                                    : BuiltinScope.REQUEST.getName())
+                            .done();
+                }
+            }
+        }));
+    }
+
+    private Set<DotName> findSubresources(IndexView index, Map<DotName, ClassInfo> scannedResources) {
+        // Identify sub-resource candidates  
+        Set<DotName> subresources = new HashSet<>();
+        for (DotName annotation : METHOD_ANNOTATIONS) {
+            Collection<AnnotationInstance> annotationInstances = index.getAnnotations(annotation);
+            for (AnnotationInstance annotationInstance : annotationInstances) {
+                DotName declaringClassName = annotationInstance.target().asMethod().declaringClass().name();
+                if (scannedResources.containsKey(declaringClassName)) {
+                    // Skip resource classes
+                    continue;
+                }
+                subresources.add(declaringClassName);
+            }
+        }
+        // Collect sub-resource locator return types
+        Set<DotName> subresourceLocatorTypes = new HashSet<>();
+        for (ClassInfo resourceClass : scannedResources.values()) {
+            ClassInfo clazz = resourceClass;
+            while (clazz != null) {
+                for (MethodInfo method : clazz.methods()) {
+                    if (method.hasAnnotation(ResteasyDotNames.PATH)) {
+                        subresourceLocatorTypes.add(method.returnType().name());
+                    }
+                }
+                if (clazz.superName().equals(DotNames.OBJECT)) {
```

Comment:
Does this `if / else` perhaps need to be inverted?

Reply:
Good catch! Thanks.